File: src/LightNap.Core/Users/Dto/Request/SearchClaimsRequestDto.cs
using LightNap.Core.Api;

namespace LightNap.Core.Users.Dto.Request
{
    /// <summary>
    /// Represents a request to search claims.
    /// </summary>
    public class SearchClaimsRequestDto : PaginationRequestDtoBase
    {
        /// <summary>
        /// Filter by user.
        /// </summary>
        public string? UserId { get; set; }

        /// <summary>
        /// Filter by claim type.
        /// </summary>
        public string? Type { get; set; }

        /// <summary>
        /// Filter by claim value.
        /// </summary>
        public string? Value { get; set; }
    }
}

File: src/lightnap-ng/src/app/core/api/dtos/users/request/search-claims-request-dto.ts
import { PagedRequestDto } from "../../paged-request-dto";


/**
 * Represents a request to search for claims with optional filtering and pagination.
 *
 * @extends PagedRequestDto
 *
 * @property {string} [type] - The username substring to search for.
 * @property {string} [value] - The value to search for in the claims.
 */
export interface SearchClaimsRequestDto extends PagedRequestDto {
   /**
     * The type to filter claims by.
     * @type {string}
     */
    type?: string;

    /**
     * The value to filter claims by.
     * @type {string}
     */
    value?: string;
}
